### DISCUSSION TITLE
identify community nodes in address book

### DISCUSSION THREAD

**Discussion by kimbor** [hiero-ledger/hiero-improvement-proposals#691] [Edits]
Add a flag to the address book to indicate whether a node belongs to a council member or a community member. Update related protobuf message and mirror node response objects.

https://github.com/hashgraph/hedera-improvement-proposal/pull/690

**Comment by steven-sheehy** [comment]
The `nodeAddress` link is broken.

**Reply by kimbor** [reply]
Thanks. I opened a PR to fix it. https://github.com/hashgraph/hedera-improvement-proposal/pull/697

**Reply by kimbor** [reply]
That PR has been merged.

**Comment by steven-sheehy** [comment]
We should use an enum value instead of a boolean `communityNode` flag to allow future expansion. Something like:

```proto
enum NodeType {
  COUNCIL = 0;
  PERMISSIONED = 1;
  PERMISSIONLESS = 2;
}
```

**Reply by kimbor** [reply]
Working on it. I'll have a PR up shortly.

**Reply by kimbor** [reply]
https://github.com/hashgraph/hedera-improvement-proposal/pull/699

**Comment by steven-sheehy** [comment]
Please add the raw protobuf changes to the HIP. This should include proto comments that note that default value. Please also show the field being added to the output of the `/api/v1/network/nodes` JSON in an example.

**Reply by kimbor** [reply]
This PR adds the raw protobuf:
https://github.com/hashgraph/hedera-improvement-proposal/pull/699

**Comment by steven-sheehy** [comment]
> Node addresses are configured in a JSON-formatted network file defined by the NodeAddressBook protobuf.

This is not really true. There is a separate JSON formatted address book used internally by the platform layer. The proto `NodeAddressBook` is separate and not used by the platform. I'm not sure if yahcli takes in JSON to update the `NodeAddressBook`, but that's irrelevant and we should just remove the JSON comment to reduce confusion.

**Reply by kimbor** [reply]
Thanks. I deleted that sentence in PR 699.

**Comment by mthometech** [comment]
Is there something different or lesser about community nodes vs council nodes? If they are the same, I would instead suggest a memo or alias field that allows any node to associate the name of the organization or individual operating the node. This would also accommodate for the future when anyone can run a node.
